scheduler-workflow/workflow.go

```diff
@@ -0,0 +1,225 @@
+
+	client, err := dapr.NewClient()
+	if err != nil {
+		panic(err)
+	}
+	defer client.Close()
+
+	daprService := http.NewService(appPort)
+	go func() {
+		log.Println("Starting app service...")
+		if err := daprService.Start(); err != nil && err.Error() != http2.ErrServerClosed.Error() {
+			log.Fatalf("error starting app service: %v", err)
+		}
+	}()
+
+	worker, err := workflow.NewWorker()
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Workflow worker initialized")
+
+	defer func() {
+		if err := worker.Shutdown(); err != nil {
+			log.Printf("Failed to shutdown workflow worker: %v", err)
+		}
+	}()
+
+	if err := worker.RegisterWorkflow(TestWorkflow); err != nil {
+		log.Fatalf("Failed to register workflow: %v", err)
+	}
+	if err := worker.RegisterActivity(TestActivity); err != nil {
+		log.Fatalf("Failed to register activity: %v", err)
+	}
+
+	// Start workflow runner
+	if err := worker.Start(); err != nil {
+		log.Fatalf("Failed to start worker: %v", err)
+	}
+	fmt.Println("Workflow worker started")
+
+	go startLonghaulWorkflow(ctx, client)
+	waitForShutdown(cancel)
+}
+
+func TestWorkflow(ctx *workflow.WorkflowContext) (any, error) {
+	var input int
+	if err := ctx.GetInput(&input); err != nil {
+		return nil, err
+	}
+	var output string
+	if err := ctx.CallActivity(TestActivity, workflow.ActivityInput(input)).Await(&output); err != nil {
+		return nil, err
+	}
+
+	err := ctx.WaitForExternalEvent("testEvent", time.Second*60).Await(&output)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := ctx.CallActivity(TestActivity, workflow.ActivityInput(input)).Await(&output); err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func TestActivity(ctx workflow.ActivityContext) (any, error) {
+	var input int
+	if err := ctx.GetInput(&input); err != nil {
+		return "", err
+	}
+
+	if stage >= 100 {
+		stage = 0
+	}
+
+	stage++
+	return fmt.Sprintf("Stage: %d", stage), nil
+}
+
+// startLonghaulWorkflow performs the following operations on a workflow:
+// start, pause, resume, raise event, terminate, purge
+func startLonghaulWorkflow(ctx context.Context, client dapr.Client) {
+	i := 0
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			fmt.Printf("Starting workflow iteration %d\n", i)
+			instanceID := fmt.Sprintf("longhaul-instance-%d", i)
+			workflowReq := &dapr.StartWorkflowRequest{
+				InstanceID:        instanceID,
+				WorkflowComponent: "",
+				WorkflowName:      "TestWorkflow",
+				Input:             i,
+				SendRawInput:      false,
+			}
+
+			respStart, err := client.StartWorkflowBeta1(ctx, workflowReq)
+			if err != nil {
+				log.Printf("Iteration %d: Failed to start workflow: %v\n", i, err)
+				continue
+			}
+			fmt.Printf("Workflow started with ID: '%s'\n", respStart.InstanceID)
+
+			err = client.PauseWorkflowBeta1(ctx, &dapr.PauseWorkflowRequest{
+				InstanceID:        instanceID,
+				WorkflowComponent: "",
+			})
+			if err != nil {
+				log.Fatalf("Failed to pause workflow: %v\n", err)
```

Comment:
We're logging the error on StartWorkflow, but exiting on this one. 
I can see the benefit of exiting the app and noticing the container getting restarted in the Grafana dashboard, but I feel like we're already testing this in our other test suites. I'd prefer the app run for as long as possible, meaning we should log the errors, but not exit.